fhir-persistence-schema/src/main/resources/db2/add_any_resource.sql

```diff
@@ -93,9 +96,9 @@ BEGIN
       -- we created the logical resource and therefore we already own the lock. So now we can
       -- safely create the corresponding record in the resource-type-specific logical_resources table
       PREPARE stmt FROM
-         'INSERT INTO ' || v_schema_name || '.' || p_resource_type || '_logical_resources (mt_id, logical_resource_id, logical_id, is_deleted, last_updated, version_id) '
-      || '     VALUES (?, ?, ?, ?, ?, ?)';
-      EXECUTE stmt USING {{ADMIN_SCHEMA_NAME}}.sv_tenant_id, v_logical_resource_id, p_logical_id, p_is_deleted, p_last_updated, p_version;
+         'INSERT INTO ' || v_schema_name || '.' || p_resource_type || '_logical_resources (mt_id, logical_resource_id, logical_id, is_deleted, last_updated, version_id, current_resource_id) '
+      || '     VALUES (?, ?, ?, ?, ?, ?, ?)';
+      EXECUTE stmt USING {{ADMIN_SCHEMA_NAME}}.sv_tenant_id, v_logical_resource_id, p_logical_id, p_is_deleted, p_last_updated, p_version, v_resource_id;
```

Comment:
interesting that we didn't set current_resource_id before... good find.

Reply:
Yeah, we weren't setting it because we didn't establish the resource_id value until later in the procedure and it just felt wrong creating a relationship to something which didn't actually exist yet (because we haven't inserted the xx_resources row). But because the id value is obtained from a sequence, we can just grab the value early and use it right away when we do the very first insert into xx_logical_resources.